docs/get-started/dev-big-picture.qmd

```diff
@@ -0,0 +1,150 @@
+---
+title: "The big picture: Builder"
+jupyter:
+  kernelspec:
+    display_name: Python 3 (ipykernel)
+    language: python
+    name: python3
+---
+
+While the "Basic Use" section covered how to configure and build a site with quartodoc, this section focuses on using quartodoc as a python program.
+
+Programming with quartodoc will help with debugging, tinkering, and extending things.
+
+## Overview
+
+When a user runs `quartodoc build`, they 
+
+* Create a [Builder](`quartodoc.Builder`) object, with the quartodoc config loaded as a [layout.Layout](`quartodoc.layout.Layout`).
+* Use [blueprint](`quartodoc.blueprint`) to process the layout into a plan for building the website.
+* Use [collect](`quartodoc.collect`) to get pages to render, and info on where documented objects live.
+
+This page will cover the basics of the Builder and this process.
+
+## The Builder
+
+The code below shows a Builder object being loaded from a `_quarto.yml` config (loaded as a python dictionary).
+
+```{python}
+import yaml
+
+from quartodoc import Builder, preview, blueprint, collect, MdRenderer
+
+cfg = yaml.safe_load("""
+quartodoc:
+  package: quartodoc
+  style: pkgdown
+  sections:
+    - title: "Some section"
+      desc: "Some description"
+      contents:
+        - name: MdRenderer
+          members: ["render", "summarize"]
+          children: separate
+""")
+
+builder = Builder.from_quarto_config(cfg)
+builder
+```
+
+Note that .from_quarto_config used the `style:` field to decide which Builder to create
+(in this case, PkgdownBuilder).
+
+We can view the config as a [layout.Layout](`quartodoc.layout.Layout`), by looking at the `.layout` attribute.
+
+```{python}
+builder.layout
+```
+
+This can be a bit difficult to read, so quartodoc implements a [preview](`quartodoc.preview`) function, which spaces things out.
+
+```{python}
+preview(builder.layout)
+```
+
+Notice the following:
+
+* `preview` represents calls like `Layout()` with a box to the left, and then
+a pipe connecting it to each of its arguments.
+* The content entry `MdRenderer` is represented by an [Auto](`quartodoc.Auto`) class.
+  This specifies a python object to look up and document.
+
+We can follow the path in the preview above, to pull out just the contents piece:
+
+```{python}
+content = builder.layout.sections[0].contents
+preview(content)
+```
+
+Next, we'll look at `blueprint()`, which processes the layout, including transforming `Auto` objects into more concrete instructions.
+
+## From config to blueprint
+
+The code below shows how `blueprint()` transforms the `Auto` entry for `MdRenderer`.
+
+```{python}
+bp = blueprint(builder.layout)
+bp_contents = bp.sections[0].contents
```

Comment:
are you only using the first sections item for brevity, or is that for a different reason?

Reply:
I think for brevity, since the printouts get very big 😭